flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxyTest.java

```diff
@@ -151,6 +190,45 @@ public void testGetShardList() throws Exception {
 						expectedStreamShard.toArray(new StreamShardHandle[actualShardList.size()])));
 	}
 
+	@Test
+	public void testGetShardListRetry() throws Exception {
```

Comment:
Would it make sense to also have a test where we exceed the number of configured retries? In that case we should not get any result.

Reply:
Good point, expanded the test to cover this.